docs/commands/lint.md

```diff
@@ -5,24 +5,55 @@ title: modular lint
 
 # `modular lint [options] [regexes...]`
 
-`modular lint` will check the diff between the current branch and your remote
-origin default branch (i.e. `master` or `main`) and only lint the `.ts`, `.tsx`,
-`.js`, `.jsx` files that have changes.
+`modular lint` without arguments or options will check the diff between the
+current branch and your remote origin default branch (i.e. `master` or `main`)
+and only lint the `.ts`, `.tsx`, `.js`, `.jsx` files that have changes. When in
+CI or when `--all` is provided, it will default to lint the entire codebase.
+Wneh `--staged` is provided, it will only lint the files staged on git.
+
+Alternatively, when it is invoked with one or more selective options
+(`--packages`, `--ancestors`, `--descendants` or `--changed`), it will lint all
+the source code files in the provided `--packages`, optionally agumenting the
+set of packages by adding their ancestors (`--ancestors`), their descendants
+(`--descendants`) or the files that have changed (`--changed`), calculated
+comparing the current state of the git repository with the branch specified by
+`compareBranch` or, if `compareBranch` is not set, with the default branch.
 
 It uses your project eslint config to lint and jest runner to report on the lint
 status of each file (`PASS` or `FAIL`). If a file has a lint warning, the
 command will be marked as failed, with printed eslint errors for each
 warning/error.
 
-When in CI, it will default to lint the entire codebase. When not in CI, it will
```

Comment:
I guess you removed this because it depends on what selective options they use (or none at all). However, without looking at the code, I'm guessing that `!!CI` mattered: this is also the case in typecheck. If I'm right, that behaviour needs to be preserved, right?

Reply:
(Assuming that behaviour *is* preserved, I guess it should be described here)

Reply:
I removed it because it was misleading and incomplete (it ignored `--regex` and `--staged`). I'll rewrite it.

Reply:
Done.